### PR TITLE
[FEAT] Update Checkin Reminder process, Send checkin reminders  to staff via the web app

### PR DESCRIPTION
**Pull request by Talleyrand333**
## Is this a Feature, Chore or Bug?
- [*] Feature
- [] Chore
- [] Bug

## Clearly and concisely describe the feature, chore or bug.
 Update Checkin Reminder process, Send checkin reminders  to staff via the web app

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.
Updated the checkin process to integrate employee checkin reminders
## Is there a business logic within a doctype?
    - [] Yes
    - [*] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.
Checkin Reminder

Screenshot 2024-07-08 at 7 25 56 PM

Supervisor Checkin Reminder
Screenshot 2024-07-09 at 10 27 28 AM
Supervisor Checkout Reminder
Screenshot 2024-07-09 at 10 29 21 AM
Checkout Reminder
Screenshot 2024-07-09 at 10 38 15 AM

## Areas affected and ensured
List out the areas affected by your code changes.
Mobile web App

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.
Push Notifications will be sent to users in the mobile web

## Did you test with the following dataset?
- [] Existing Data
- [*] New Data

## Was child table created? NO
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [*] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [*] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/api/tasks.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
-import itertools
-from datetime import timedelta
+import itertools,time
```
Comment (ks093) [inline one_fm/api/tasks.py]: @Talleyrand333 Is time being used anywhere?

File: one_fm/api/tasks.py [hunk 2]
```diff
@@ -1981,6 +1997,10 @@ def initiate_checkin_notification(res):
 			""").format(recipient=recipient, shift=recipient.operations_shift, date=cstr(now_time), employee=recipient.employee, time=str(recipient.start_datetime))
 			if action_user is not None: #and not has_checkin_record(recipient.employee, recipient.log_type, res.date):
 				send_notification(title, subject, action_message, category, [action_user])
+			if action_user_id:
```
Comment (ks093) [inline one_fm/api/tasks.py:2001]: @Talleyrand333 There is a possibility of local variable referenced before assignment error with this condition.

File: one_fm/api/tasks.py [hunk 3]
```diff
@@ -2032,6 +2054,10 @@ def initiate_checkin_notification(res):
 			""").format(recipient=recipient, shift=recipient.operations_shift, date=cstr(now_time), employee=recipient.employee, time=str(recipient.start_datetime))
 			if action_user is not None:# and not has_checkin_record(recipient.employee, recipient.log_type, res.date):
 				send_notification(title, subject, action_message, category, [action_user])
+			if action_user_id:
```
Comment (ks093) [inline one_fm/api/tasks.py:2059]: @Talleyrand333 Same here. There is a possibility of local variable referenced before assignment error with this condition.

File: one_fm/api/tasks.py (hunk 1 above)
Comment (Talleyrand333) [reply one_fm/api/tasks.py]: @ks093 thanks for pointing this out, It was used to test how long the process takes. it has been removed

File: one_fm/api/tasks.py (hunk 3 above)
Comment (Talleyrand333) [reply one_fm/api/tasks.py:2059]: @ks093  thanks, I have declared it at the start the the loop for each.

**Review by ks093 — APPROVED** [review]

**Pull request merged**